vnext/local-cli/generator-windows/templates/cs/proj/MyApp.csproj

```diff
@@ -154,25 +153,33 @@
     </PackageReference>
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="$(ReactNativeWindowsDir)\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj">
-      <Project>{f7d32bd0-2749-483e-9a0d-1635ef7e3136}</Project>
-      <Name>Microsoft.ReactNative</Name>
-    </ProjectReference>
-    <PackageReference Include="Microsoft.UI.Xaml">
-      <Version>2.3.191129002</Version>
-    </PackageReference>
+    <Reference Include="Microsoft.UI.Xaml, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null, ContentType=WindowsRuntime">
+      <HintPath>..\packages\Microsoft.UI.Xaml.2.3.191129002\lib\uap10.0\Microsoft.UI.Xaml.winmd</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.UI.Xaml.Design, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
+      <HintPath>..\packages\Microsoft.UI.Xaml.2.3.191129002\lib\uap10.0\Design\Microsoft.UI.Xaml.Design.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
   </ItemGroup>
-  <Import Project="$(ReactNativeWindowsDir)\Microsoft.ReactNative.SharedManaged\Microsoft.ReactNative.SharedManaged.projitems" Label="Shared" />
   <PropertyGroup Condition=" '$(VisualStudioVersion)' == '' or '$(VisualStudioVersion)' &lt; '16.0' ">
     <VisualStudioVersion>16.0</VisualStudioVersion>
   </PropertyGroup>
   <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />
-  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.props" />
-  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.targets" />
+  <ImportGroup Label="ReactNativeTargets">
+    <Import Project="..\..\node_modules\react-native-windows\PropertySheets\External\Microsoft.ReactNative.CSharpApp.targets" />
+    <Import Project="AutolinkedNativeModules.g.targets" Condition="Exists('AutolinkedNativeModules.g.targets')" />
+  </ImportGroup>
   <PropertyGroup>
     <PreBuildEvent>
     </PreBuildEvent>
   </PropertyGroup>
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105.The missing file is {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('..\packages\Microsoft.UI.Xaml.2.3.191129002\build\Microsoft.UI.Xaml.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.UI.Xaml.2.3.191129002\build\Microsoft.UI.Xaml.props'))" />
```

Comment:
move to targets

Reply:
no, this will break upgrades

Reply:
When we can switch to use PackageReference this will no longer be needed.

---
In reply to: [421149749](https://github.com/microsoft/react-native-windows/pull/4816#discussion_r421149749) [](ancestors = 421149749)